external/kafka/src/main/scala/org/apache/gearpump/streaming/kafka/lib/KafkaConfig.scala

```diff
@@ -30,42 +33,29 @@ object KafkaConfig {
 
   val NAME = "kafkaConfig"
 
-  // consumer config
-  val ZOOKEEPER_CONNECT = "kafka.consumer.zookeeper.connect"
   val CONSUMER_TOPICS = "kafka.consumer.topics"
-  val SOCKET_TIMEOUT_MS = "kafka.consumer.socket.timeout.ms"
-  val SOCKET_RECEIVE_BUFFER_BYTES = "kafka.consumer.socket.receive.buffer.bytes"
-  val CLIENT_ID = "kafka.consumer.client.id"
-  val FETCH_MESSAGE_MAX_BYTES = "kafka.consumer.fetch.message.max.bytes"
   val CONSUMER_EMIT_BATCH_SIZE = "kafka.consumer.emit.batch.size"
   val FETCH_THRESHOLD = "kafka.consumer.fetch.threshold"
   val FETCH_SLEEP_MS = "kafka.consumer.fetch.sleep.ms"
-
-  // producer config
   val PRODUCER_TOPIC = "kafka.producer.topic"
-  val PRODUCER_BOOTSTRAP_SERVERS = "kafka.producer.bootstrap.servers"
-  val PRODUCER_ACKS = "kafka.producer.acks"
-  val PRODUCER_BUFFER_MEMORY = "kafka.producer.buffer.memory"
-  val PRODUCER_COMPRESSION_TYPE = "kafka.producer.compression.type"
   val PRODUCER_BATCH_SIZE = "kafka.producer.batch.size"
-  val PRODUCER_RETRIES = "kafka.producer.retries"
-
-  // storage config
   val STORAGE_REPLICAS = "kafka.storage.replicas"
-
-  // grouper config
   val GROUPER_FACTORY_CLASS = "kafka.grouper.factory.class"
-
-  // task config
   val MESSAGE_DECODER_CLASS = "kafka.task.message.decoder.class"
   val TIMESTAMP_FILTER_CLASS = "kafka.task.timestamp.filter.class"
 
-  def apply(config: Config): KafkaConfig = new KafkaConfig(config)
+  def apply(gearpumpConfig: Config): KafkaConfig = new KafkaConfig(gearpumpConfig)
+
+  def apply(resource: String): KafkaConfig = KafkaConfig(ConfigFactory.parseResources(resource))
+
+  def apply(): KafkaConfig = KafkaConfig(ConfigFactory.empty())
 
   private val LOG: Logger = LogUtil.getLogger(getClass)
 }
 
-class KafkaConfig(config: Config) extends Serializable  {
+class KafkaConfig(gearpumpConfig: Config,
```

Comment:
if so, please add kafka as prefix or postfix. GearpumpConfig is a too general term.